### PR TITLE
modified weighted score and primary score metric

### PR DESCRIPTION
**Pull request by ikeola13**
## Description

Modified weighted score and primary score metric

## How Has This Been Tested?

-modified existing tests

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)
- [x] The code follows our [Best Practices](/docs/best-practices.md) (React coding standards, PatternFly usage, performance considerations)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

## Summary by CodeRabbit

* **New Features**
  * Collection-based evaluation requests now include per-benchmark configuration (id, provider, primary score, pass criteria).

* **Bug Fixes**
  * Evaluation score display now consistently shows the overall evaluation score.
  * Benchmark indices are correctly aligned/matched for collection-based evaluations.

* **Tests**
  * Increased test coverage and adjusted tests to validate collection benchmark merging and request payloads.

### CODE REVIEW

File: packages/eval-hub/frontend/src/app/pages/EvaluationResultsPage.tsx [hunk 1]
```diff
@@ -113,11 +112,8 @@ const EvaluationResultsPage: React.FC = () => {
     if (!job) {
       return '-';
     }
-    if (selectedBenchmark && benchmarks.length > 1) {
-      return getBenchmarkResultScore(job, selectedBenchmark.id, selectedBenchmark.benchmark_index);
-    }
     return getResultScore(job);
-  }, [job, selectedBenchmark, benchmarks.length]);
+  }, [job]);
```
Comment (pnaik1) [inline packages/eval-hub/frontend/src/app/pages/EvaluationResultsPage.tsx:116]: Benchmark cards still render with isSelected state and fire click events — they still drive BenchmarkResultDetails and MLflow tabs — but clicking them no longer updates the headline "Evaluation score". Cards look interactive but the most prominent metric doesn't respond to selection.

**Review by pnaik1 — CHANGES_REQUESTED** [review]
Great work here @ikeola13 some nits

File: packages/eval-hub/frontend/src/app/utilities/evaluationUtils.ts [hunk 2]
```diff
@@ -9,11 +9,31 @@ export const getJobBenchmarks = (job: EvaluationJob): NonNullable<EvaluationJob[
...
   }
   if (job.collection?.benchmarks?.length) {
-    return job.collection.benchmarks;
+    const collBenchmarks = job.collection.benchmarks;
+    const resultsBenchmarks = job.results.benchmarks;
+    if (resultsBenchmarks?.length) {
+      /* eslint-disable camelcase */
+      return resultsBenchmarks.map((rb) => {
+        const config = collBenchmarks.find(
+          (cb, idx) => cb.id === rb.id && (cb.benchmark_index ?? idx) === rb.benchmark_index,
```
Comment (pnaik1) [inline packages/eval-hub/frontend/src/app/utilities/evaluationUtils.ts]: If rb.benchmark_index is undefined, then number === undefined is always false → config is always undefined → primary_score/pass_criteria silently fail to merge. The new tests only cover explicit indices and miss this edge case.
Fix:

**Review by pnaik1 — CHANGES_REQUESTED** [review]
